test/utils.py

```diff
@@ -528,10 +526,17 @@ def _mongo_client(host, port, authenticate=True, directConnection=None,
         client_options['directConnection'] = directConnection
     client_options.update(kwargs)
 
-    client = MongoClient(_connection_string(host, authenticate), port,
-                         **client_options)
-
-    return client
+    uri = _connection_string(host)
+    if client_context.auth_enabled and authenticate:
+        # Only add the default username or password if one is not provided.
+        res = parse_uri(uri)
+        if (not res['username'] and not res['password'] and
+                'username' not in client_options and
+                'password' not in client_options):
+            client_options['username'] = db_user
+            client_options['password'] = db_pwd
+
+    return MongoClient(uri, port, **client_options)
```

Comment:
These changes were required in order to support passing `mongodb+srv` addresses through rs_client() and friends.